### PR TITLE
EQL: endsWith function implementation

### PR DESCRIPTION
**Pull request by astefan**
Addresses https://github.com/elastic/elasticsearch/issues/53854

### CODE REVIEW

**Review by aleksmaus — APPROVED** [review]
one nit, otherwise LGTM

File: x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/expression/function/scalar/string/EndsWithFunctionProcessor.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            throw new EqlIllegalArgumentException("A string/char is required; received [{}]", source);
+        }
+        if (pattern == null) {
+            return null;
+        }
+        if (pattern instanceof String == false && pattern instanceof Character == false) {
+            throw new EqlIllegalArgumentException("A string/char is required; received [{}]", pattern);
+        }
+
+        return source.toString().toLowerCase(Locale.ROOT).endsWith(pattern.toString().toLowerCase(Locale.ROOT));
```
Comment (rw-access) [inline x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/expression/function/scalar/string/EndsWithFunctionProcessor.java]: based off some of the conversations that have been happening, I think we're going to outsource this to something else eventually.
I don't know if that  means we should do case-sensitivity in all PRs now, or come back to this later

Comment (aleksmaus) [reply x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/expression/function/scalar/string/EndsWithFunctionProcessor.java]: IMHO, this covers the feature parity with the original EQL implementation, and passes test against the EQL test data/queries. 
We can merge this as is and keep iterating and moving forward.
Also this gives us a chance to refactor some of the code that can be better shared/reused across all of the functions implementations better.

**Review by costin — APPROVED** [review]
LGTM.

The string functions have a lot of boilerplate in common and after this PR gets in, it's worth revisiting and extracting a common class for them.

**Pull request merged**